presto-main/src/main/resources/webapp/src/components/QueryDetail.jsx

```diff
@@ -892,30 +911,31 @@ export class QueryDetail extends React.Component {
             return;
         }
 
-        const tasks = this.getTasksFromStage(this.state.lastSnapshotTasks).filter(task => this.state.taskFilter(task.taskStatus.state), this);
+        const tasks = this.getTasksFromStage(this.state.lastSnapshotTasks).filter(task => this.state.taskFilter.predicate(task.taskStatus.state), this);
```

Comment:
Can you add a special when `taskFilter` is `TASK_FILTER.NONE`, so the component don't even iterate over all tasks? Ideally it would not fetch task info from the coordinator but i'm not sure how much we would gain from optimizing further like that.

Reply:
Yeah, that makes sense. Let me add a special case.